pysam/libcutils.pyx

```diff
@@ -82,7 +82,29 @@ cpdef qualities_to_qualitystring(qualities, int offset=33):
 
 
 ########################################################################
+## String encoding configuration facilities
 ########################################################################
+
+# Codec error handler that just interprets each bad byte as ISO-8859-1.
+def latin1_replace(exception):
+    return (chr(exception.object[exception.start]), exception.end)
+
+
+cdef str ERROR_HANDLER = 'strict'
+
+cpdef get_encoding_error_handler():
+    return ERROR_HANDLER
+
+cpdef set_encoding_error_handler(name):
+    global ERROR_HANDLER
+    previous = ERROR_HANDLER
+    if name.startswith('pysam.'):
```

Comment:
Is this check intended to be more general?  It isn't obvious why you're only checking the prefix.  (Feel free to tell me to hold my horses until you're ready for it to be reviewed.)

Reply:
The user will be calling `set_encoding_error_handler()` with either a standard handler, one they wrote themselves, or one that pysam conveniently supplies. We only need to register the pysam-supplied ones if the user is actually asking for a pysam-supplied one, and in the interests of “libraries shouldn't do weird stuff if they don't need to” this check avoids registering ours if the user is using a standard one or their own.

The check could be `name == 'pysam.latin1replace'` but I'm imagining the list might expand in future. In particular, I might add a `pysam.latin15replace` as well…

That was the idea anyway… I suppose once we add file class `set_encoding_error_handler()` methods as well, it would make more sense to move this out of the global `set_encoding_error_handler` function and just always register them in whatever the appropriate form of `__init__` processing is.

Reply:
That makes perfect sense now that I understand where the concept is/was/may still be going.

Reply:
I've changed it so that it always just registers the handler name when _libcutils.pyx_ is loaded. (I'm no expert, but it seems like just putting the code at the top-level of the _.pyx_ file so it is executed on load is “the right form of `__init__` processing” in this context…?)

This paves the way for future file class `set_encoding_error_handler()` methods (e.g. for HTSFile) — if user code was using those, then the global `set_encoding_error_handler()` function would likely never be called (so it would be a bad place to do this init processing!).